moto/cloudformation/responses.py

```diff
@@ -154,11 +154,36 @@ def stack_name_exists(self, new_stack_name: str) -> bool:
                 return True
         return False
 
+    def validate_template_and_stack_body(self) -> None:
+        if (
+            self._get_param("TemplateBody") or self._get_param("TemplateURL")
+        ) and self._get_param("UsePreviousTemplate", "false").lower() == "true":
+            raise ValidationError(
+                message="An error occurred (ValidationError) when calling the CreateChangeSet operation: You cannot specify both usePreviousTemplate and Template Body/Template URL."
+            )
+        elif (
+            not self._get_param("TemplateBody")
+            and not self._get_param("TemplateURL")
+            and self._get_param("UsePreviousTemplate", "false").lower() == "false"
+        ):
+            raise ValidationError(
+                message="An error occurred (ValidationError) when calling the CreateChangeSet operation: Either Template URL or Template Body must be specified."
+            )
+
     def create_change_set(self) -> str:
         stack_name = self._get_param("StackName")
         change_set_name = self._get_param("ChangeSetName")
         stack_body = self._get_param("TemplateBody")
         template_url = self._get_param("TemplateURL")
+        if self._get_param("ChangeSetType", "CREATE") == "UPDATE":
+            use_previous_template = (
+                self._get_param("UsePreviousTemplate", "false").lower() == "true"
+            )
```

Comment:
```suggestion
        update_or_create = self._get_param("ChangeSetType", "CREATE")
        use_previous_template = (
                self._get_param("UsePreviousTemplate", "false").lower() == "true"
            )
        if update_or_create == "UPDATE":
```
The `update_or_create` variable is already created a few lines below. It would be cleaner to declare the variable once in the beginning, and then use it everywhere

Note that I'm only suggesting this to give an idea - it will probably fail our linter if we push this change as-is